weaviate/collection/data.py

```diff
@@ -61,6 +61,41 @@ def _insert(self, weaviate_obj: Dict[str, Any]) -> uuid_package.UUID:
             pass
         raise UnexpectedStatusCodeException("Creating object", response)
 
+    def _insert_many(self, objects: List[DataObject]) -> _BatchReturn:
+        weaviate_objs: List[weaviate_pb2.BatchObject] = [
+            weaviate_pb2.BatchObject(
+                class_name=self.name,
+                vector=obj.vector if obj.vector is not None else None,
+                uuid=str(obj.uuid) if obj.uuid is not None else str(uuid_package.uuid4()),
+                properties=self.__parse_properties_grpc(obj.data),
+                tenant=self._tenant,
+            )
+            for obj in objects
+        ]
+
+        errors = self._batch.batch(weaviate_objs)
+
+        all_responses: List[Union[uuid_package.UUID, Error]] = []
+        return_success: Dict[int, uuid_package.UUID] = {}
+        return_errors: Dict[int, Error] = {}
+
+        for idx, obj in enumerate(weaviate_objs):
```

Comment:
I'm not sure if this is faster than 2 seperate loops with comprehension. Here you need to extend the list in every iteration which is slow:
https://stackoverflow.com/questions/30245397/why-is-a-list-comprehension-so-much-faster-than-appending-to-a-list

Reply:
Oh how interesting, I was unaware of this? I've pushed a change that may mitigate this, let me know if you think it does!